src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1

```diff
@@ -0,0 +1,765 @@
+        # do not suport.
+        $supportedFeatures = @("proxy")
+        if ($arcAgentrySettings) {
+            foreach ($key in $arcAgentrySettings.Keys) {
+                if (-not $supportedFeatures.Contains($key.ToLower())) {
+                    Write-Warning "Arc Agentry feature '${key}' is not supported for Connected Kubernetes"
+                }
+            }
+        }
+        if ($arcAgentryProtectedSettings) {
+            foreach ($key in $arcAgentryProtectedSettings.Keys) {
+                if (-not $supportedFeatures.Contains($key.ToLower())) {
+                    Write-Warning "ArcAgentry feature '${key}' is not supported for Connected Kubernetes"
+                }
+            }
+        }
+        #Endregion
+
+        # XW TODO: helm install can be extracted as common function to be used by both Set and New
+        #Region check helm install
+        try {
+            Set-HelmClientLocation
+            $HelmVersion = helm version --template='{{.Version}}' --kubeconfig $KubeConfig
+            if ($HelmVersion.Contains("v2")) {
+                Write-Error "Helm version 3+ is required (not ${HelmVersion}). Learn more at https://aka.ms/arc/k8s/onboarding-helm-install"
+                return
+            }
+            $HelmVersion = helm version --short --kubeconfig $KubeConfig
+
+            # Compare the helm version to 3.8 in a symantic versioning valid way
+            # Strip the leading "v" from the helm version and discard any metadata
+            $HelmVersion = $HelmVersion.Substring(1)
+            $HelmVersion = $HelmVersion.Split('+')[0]
+            $helmV380 = [System.Version]::Parse("3.8.0")
+            $helmThisVersion = [System.Version]::Parse($HelmVersion)
+            if ($helmThisVersion -lt $helmV380) {
+                Write-Error "Helm version of at least 3.8 is required for latest OCI handling."
+                Return
+            }
+        }
+        catch {
+            throw "Failed to install Helm version 3+ ($_). Learn more at https://aka.ms/arc/k8s/onboarding-helm-install"
+        }
+        #EndRegion
+        $helmClientLocation = 'helm'
+
+        # XW TODO: can be extracted as common function to be used by both Set and New
+        #Region get release namespace
+        $ReleaseInstallNamespace = Get-ReleaseInstallNamespace
+        $ReleaseNamespace = $null
+        try {
+            # !!PDS: Seems like this showing "Error: Release: Not found" is not an error but a warning that implies there just is not a release.  Can we quench this?
+            $ReleaseNamespace = (helm status azure-arc -o json --kubeconfig $KubeConfig --kube-context $KubeContext -n $ReleaseInstallNamespace 2> $null | ConvertFrom-Json).namespace
+        }
+        catch {
+            Write-Error "Fail to find the namespace for azure-arc."
+        }
+        #Endregion
+
+        # XW: different with New command
+        #Region validate release namespace
+        if ($null -ne $ReleaseNamespace) {
```

Comment:
This probably wants to be `if ('' -ne -$ReleaseNameSpace) {` or the "check for null or empty string" function that you found.

Reply:
Done :)